examples/seismic/acoustic/wavesolver.py

```diff
@@ -108,12 +111,24 @@ def forward(self, src=None, rec=None, u=None, model=None, save=None, **kwargs):
         model = model or self.model
         # Pick vp from model unless explicitly provided
         kwargs.update(model.physical_params(**kwargs))
+        # Get the operator
+        op_fwd = self.op_fwd(save=save, factor=factor)
+        # Prepare parameters for operator apply
+        op_args = {'src': src, 'rec': rec, 'u': u, 'dt': kwargs.pop('dt', self.dt)}
+        op_args.update(kwargs)
 
         # Execute operator and return wavefield and receiver data
-        summary = self.op_fwd(save).apply(src=src, rec=rec, u=u,
-                                          dt=kwargs.pop('dt', self.dt), **kwargs)
-
-        return rec, u, summary
+        if factor:
```

Comment:
No, usnap needs to be create here like `u` then passed as argument

Reply:
Fixed. usnaps is created now.